### DISCUSSION TITLE
IAutorizeDataPortal and Uno Platform

### DISCUSSION THREAD

**Discussion by keithf1138** [MarimerLLC/csla#1792] [Questions]
I have a new project working on an trying out using UNO Platform along with Blazor for a tablet and a web application. They fit different needs, but have some shared objects.  So have a solution all setup with Blazor Server and have some POC functionality working. So moved on to get UNO Tablet client working.  Got the UNOExample CSLA example mostly working. Only issue is that when you start it up the PersonList form gets an OperationNotSupported exception upon the 1st attempt to talk to the DataPortal. If you immediately try again then it works, weird I know.

OK so adding Uno client to the main solution and try to run it and I get. "csla authentication provider specified does not implement iauthorizedataportal". I have gone over the UnoExample and my solution and I just cant seem to figure out what the difference is.

Any ideas?

**Comment by rockfordlhotka** [comment]
Can you provide the `ToString` of that OperationNotSupported exception? It would be useful to see the full detail on that.

**Reply by keithf1138** [reply]
Couldnt hit a break point or dump it so here is a screenshot.

![DataPortalException](https://user-images.githubusercontent.com/17910982/91609173-8791a480-e93c-11ea-824f-d39f15feb467.png)

**Reply by rockfordlhotka** [reply]
Wait - the NotSupportedException and the IAuthorizeDataPortal are the same issue? I thought we had two issues here?

**Reply by rockfordlhotka** [reply]
Maybe start a whole separate discussion for the NotSupportedException so we can keep the issues straight?

**Reply by keithf1138** [reply]
Yeah will do that. Not trying to get past that, but it is an issue others may hit.

**Comment by rockfordlhotka** [comment]
Regarding the `IAuthorizeDataPortal`, that should default to using an internal implementation of type `NullAuthorizer`.

I'm assuming you are not overriding that setting in your config, and so this leads me to believe that perhaps there's some DLL version mismatch happening.

The code throwing this exception is probably here:

```c#
      if (null == authProviderType)
        throw new ArgumentNullException(nameof(authProviderType), Resources.CslaAuthenticationProviderNotSet);
      if (!typeof(IAuthorizeDataPortal).IsAssignableFrom(authProviderType))
        throw new ArgumentException(Resources.AuthenticationProviderDoesNotImplementIAuthorizeDataPortal, nameof(authProviderType));
```
And you can see that it first checks for `null`. So the value isn't null, but also isn't implementing the interface - which seems to imply

1. You did override the value in config and are telling it to use a type that doesn't implement the interface
1. You did NOT override config, and somehow the `NullAuthorizer` being used is from a different version of the DLL (NuGet package)

**Reply by keithf1138** [reply]
Config file in which project? That looks like code. Which project and code file should that go in?

**Reply by keithf1138** [reply]
So it has to be a DLL mismatch just cant find one.

Dal - netstandard2.0,CSLA 5.3.0, 
DalSql - netstandard2.0, CSLA 5.3.0
BusinessLibrary - netstandard2.0,CSLA 5.3.0
AppServer - netcoreapp3.1,CSLA 5.3.0, CSLA.AspNetCore 5.3.0, Microsoft.Extensions.DependencyInjection 3.1.7, Microsoft.Extensions.DependencyInjection.Abstractions 3.1.7
UnoWasm - Csla 5.3.0, UNO.UI 3.0.12, UNO.UI.WebAssembly 3.0.12, UNO.UI.RemoteControl 3.0.12, UNO.Wasm.BootStrap 1.3.4, UNO.WasmBootStrap.DevServer 1.3.4

**Reply by rockfordlhotka** [reply]
That code is server-side, so the config would be in web.config or in your server's startup.cs.

**Reply by keithf1138** [reply]
I realized you were wondering. I have made no changes to the config.  Gotta be some sort of conflict. I assume between the Wasm project and the AppServer.

**Comment by keithf1138** [comment]
NotSupported is in the UnoExample project.

It happens the 1st time

var obj = await DataPortal.FetchAsync<PersonList>();

is executed. So if I put a Try catch around that and then execute it again it works.

Weird.

The IAuthorizeDataPortal is in my project.

**Reply by keithf1138** [reply]
I imagine the code that sets it up is in the ApplicationContextManager class. Also at this point I am using In-Proc AppServer.
